src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -359,6 +361,25 @@ void WiFiManager::ScanDoneHandler(uint8_t * data)
     }
 }
 
+void WiFiManager::_SendRouterSolicitation(System::Layer * layer, void * param)
+{
+    net_if * iface = InetUtils::GetInterface();
+    if (iface && iface->if_dev->link_addr.type == NET_LINK_ETHERNET)
+    {
+        net_if_start_rs(iface);
```

Comment:
The docs of `net_if_start_rs` says: "Start neighbor discovery and send router solicitation message."
https://docs.zephyrproject.org/apidoc/latest/group__net__if.html#gab9bdb7f8a9eeed4d9b70965b8af82cb7

@kkasperczyk-no do you know what does it actually mean to "start neighbor discovery"? I mean, if we really need to start it every time we send the RS? Should' we call the `net_if_stop_rs` anywhere?

Reply:
Actually no, looking at the implementation it just sends single RS packet through the given interface. Looking at the code Zephyr doesn't call net_if_stop_rs anywhere beside the case when it wants to cancel sending RS before it was done (at least that's my guess looking at the code).

Reply:
Ok, maybe the function name is just a bit misleading.

Reply:
Hmm. Actually, the function looks like sending RS a few times. Otherwise, it's hard to explain this:
```
sys_slist_append(&active_rs_timers, &ipv6->rs_node);
```
Explicit stop is not needed because `rs_timeout` calls `sys_slist_remove(&active_rs_timers, NULL, &ipv6->rs_node);`. Maybe the impl is just broken? Did you see that it doesn't send multiple RSes or you assumed so @kkasperczyk-no?

Reply:
The impl seems incorrect in sending RS every second rather than 4 seconds, but this is a different issue.

Reply:
I tested it some time ago and I saw on the wireshark, as many RS packets, as I called this method, so I assumed it is working fine.